src/types.ts

```diff
@@ -51,9 +51,8 @@ export type Event<TEvent extends EventObject> = TEvent['type'] | TEvent;
  * as well as the built in event types and/or objects.
  */
 export type OmniEvent<TEvent extends EventObject> =
-  | TEvent['type']
-  | BuiltInEvent<TEvent>['type']
-  | OmniEventObject<TEvent>;
+  | OmniEventObject<TEvent>
```

Comment:
this has fixed toEventObject calls, both versions (before and after) should be semantically equivalent, but the previous gave TS hard time (for some reason 🤷‍♂ ) to infer TEvent inside of `toEventObject`